src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -407,31 +423,25 @@ private static bool TryGetClassDeclarationList(INamedTypeSymbol typeSymbol, [Not
                 return typeGenerationSpec;
             }
 
-            internal static bool IsSyntaxTargetForGeneration(SyntaxNode node) => node is ClassDeclarationSyntax { AttributeLists: { Count: > 0 }, BaseList: { Types : {Count : > 0 } } };
+            internal static bool IsSyntaxTargetForGeneration(SyntaxNode node) => node is ClassDeclarationSyntax { BaseList: { Types : { Count : > 0 } } };
```

Comment:
For diffs between here and EOF, I'm relaxing the criteria for types we consider for generation

**Before**: include all classes that are annotated with a type called "JsonSerializableAttribute", and also derive or implement at least one type.
**After**: include all classes that derive from a type called "JsonSerializerContext".

I believe this change is fine since we'll have about the same number of types to consider in most source gen scenarios.

Reply:
Note that this could impact perf. I would test the before/after of this change and ensure for large solutions that target `net6.0` you aren't negatively impacting the perf.

I can get you a large solution offline, if you need it to test.

Reply:
It is possible this change could also improve perf, depending on the individual performance attributes vs. base classes, and of course the actual solution and its types (do they have base classes, attributes, etc).

@eerhardt how do you suggest measuring perf? Manual measuring the VS experience?

Reply:
@ericstj made a micro benchmark for source generators: https://github.com/ericstj/sample-code/tree/generatorBenchmark

But honestly, I would measure it in VS with a large solution. The way I measured the time taken was to add an EventSource around the source generator implementation. And then use PerfView to see how much time was spent in the source generator when it wasn't doing any work.

But if you are using the latest Roslyn builds (as of 5 days ago), this timing measurement is built in. See https://github.com/dotnet/roslyn/pull/55892.

Reply:
One possible way to mitigate would be check the member count of the type. In pseudocode:
```csharp
internal static bool IsSyntaxTargetForGeneration(SyntaxNode node) =>
    node is ClassDeclarationSyntax { 
            { BaseList: { Types : { Count : > 0 } } and 
            { AttributeLists: { Count: > 0 } } or 
            { MemberList: { Count: 0 } } };
```
Note that this would contradict my feedback in https://github.com/dotnet/runtime/pull/58768/files#r704458155.